libqtile/config.py

```diff
@@ -496,7 +496,7 @@ def _items(self, name: str) -> ItemT:
         elif name == "window" and self.group is not None:
             return True, [i.wid for i in self.group.windows]
         elif name == "bar":
-            return False, [x.position for x in self.gaps]
+            return False, [x.position for x in self.gaps if isinstance(x, CommandObject)]
```

Comment:
I think it would be more readable to test here and at the other 3 `isinstance` calls added below that the variable is a `Bar` directly rather than `CommandObject`, because it makes it clear to the reader that we're choosing bars and ignoring gaps